Backport of:

From fa35b2da77a3995af5bd366259e316a86c0d1362 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 9 Jul 2017 09:06:26 -0400
Subject: [PATCH] Zero pixel buffer

Avoid a data leak in case of incorrect file by clearing a buffer

bug:  https://github.com/ImageMagick/ImageMagick/issues/556
bug-debian: https://bugs.debian.org/867893
origin: https://github.com/ImageMagick/ImageMagick/commit/1737ac82b335e53376382c07b9a500d73dd2aa11

(cherry picked from commit 1737ac82b335e53376382c07b9a500d73dd2aa11)
---
 coders/jpeg.c | 2 ++
 1 file changed, 2 insertions(+)

Index: imagemagick-6.7.7.10/coders/jpeg.c
===================================================================
--- imagemagick-6.7.7.10.orig/coders/jpeg.c	2017-07-21 09:55:06.413277092 -0400
+++ imagemagick-6.7.7.10/coders/jpeg.c	2017-07-21 09:57:27.961283692 -0400
@@ -1241,6 +1241,8 @@ static Image *ReadJPEGImage(const ImageI
     jpeg_info.output_components*sizeof(JSAMPLE));
   if (jpeg_pixels == (JSAMPLE *) NULL)
     ThrowReaderException(ResourceLimitError,"MemoryAllocationFailed");
+  (void) ResetMagickMemory(jpeg_pixels,0,image->columns* 
+    jpeg_info.output_components*sizeof(*jpeg_pixels));
   /*
     Convert JPEG pixels to pixel packets.
   */
